src/mixer/basetrackplayer.cpp

```diff
@@ -335,3 +345,30 @@ void BaseTrackPlayerImpl::setupEqControls() {
     m_pSpeed = new ControlObjectSlave(group, "rate", this);
     m_pPitchAdjust = new ControlObjectSlave(group, "pitch_adjust", this);
 }
+
+void BaseTrackPlayerImpl::slotPassthroughEnabled(double v) {
+    bool configured = m_pInputEnabled->toBool();
+    bool passthrough = v > 0.0;
+
+    // Warn the user if they try to enable passthrough on a player with no
+    // configured input.
+    if (!configured && passthrough) {
+        m_pPassthroughEnabled->set(0.0);
```

Comment:
Good suggestion, though that wouldn't work because the value change request would mean this code would never run with a value that would emit the signal. The EngineDeck would have to emit a signal that BaseTrackPlayer would receive and then rebroadcast -- and that's what I want to avoid.